src/main/java/org/apache/maven/plugins/javadoc/AbstractJavadocMojo.java

```diff
@@ -6543,6 +6557,38 @@ private static String getJavadocLink( MavenProject p )
         return url + "/" + destDir;
     }
 
+    /**
+     * @param groupId not null
+     * @param artifactId not null
+     * @param version not null
+     * @return the javadoc link based on the project javadoc.io url
+     *         i.e. <code>https://www.javadoc.io/doc/${group_id}/${artifact_id}/${version_id}/package-list</code>
+     */
+    static String getJavadocIOLink( String groupId, String artifactId, String version )
+    {
+        String baseUrl = "https://javadoc.io/doc/";
+        return baseUrl + urlEncode( groupId ) + "/" + urlEncode( artifactId ) + "/" + urlEncode( version );
+    }
+
+    static String urlEncode( String s )
+    {
+        if ( s == null )

```

Comment:
don't need this check if your nonnull claims above are true. also this method should never return null